src/CabanaPD_Particles.hpp

```diff
@@ -646,6 +651,133 @@ class Particles<MemorySpace, LPS, Dimension>
     using base_type::_timer;
 };
 
+template <class MemorySpace, int Dimension>
+class Particles<MemorySpace, PMB, TemperatureDependent, Dimension>
+    : public Particles<MemorySpace, PMB, TemperatureIndependent, Dimension>
+{
+  public:
+    using self_type =
+        Particles<MemorySpace, PMB, TemperatureDependent, Dimension>;
+    using base_type =
+        Particles<MemorySpace, PMB, TemperatureIndependent, Dimension>;
+    using memory_space = typename base_type::memory_space;
+    using base_type::dim;
+
+    // Per particle.
+    using base_type::n_ghost;
+    using base_type::n_global;
+    using base_type::n_local;
+    using base_type::size;
+
+    // These are split since weighted volume only needs to be communicated once
+    // and dilatation only needs to be communicated for LPS.
+    using scalar_type = typename base_type::scalar_type;
+    using aosoa_temp_type = Cabana::AoSoA<scalar_type, memory_space, 1>;
+
+    // Per type.
+    using base_type::n_types;
+
+    // Simulation total domain.
+    using base_type::global_mesh_ext;
+
+    // Simulation sub domain (single MPI rank).
+    using base_type::ghost_mesh_hi;
+    using base_type::ghost_mesh_lo;
+    using base_type::local_mesh_ext;
```

Comment:
What is "local_mesh_ext"?

Reply:
Extent - I can rename these to be more clear later